src/components/PreCodeBlock.tsx

```diff
@@ -5,23 +5,113 @@ interface PreProps extends DetailedHTMLProps<HTMLAttributes<HTMLPreElement>, HTM
   raw?: string
 }
 
-export function PreCodeBlock ({ children, ...props }: PreProps) {
+interface ReleaseInfo {
+  id: number;
+  name: string;
+  public: boolean;
+  created_at: string;
+  promoted_at: string;
+  version: string;
+  git_sha: string;
+  is_development: boolean;
+  tags: string[] | null;
+  canonical_name: string;
+}
+
+async function getLatestVersion() {
+  const url = new URL('https://releaseregistry.sourcegraph.com/v1/releases/sourcegraph/latest');
+
+  const response = await fetch(url.toString(), {
+    method: 'GET',
+    headers: {
+      'Accept': 'application/json'
+    }
+  });
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const releaseInfo: ReleaseInfo = await response.json();
+  return releaseInfo.version
+}
+
+function trimPrefix(str: string, prefix: string) {
+  if (str.startsWith(prefix)) {
+    return str.slice(prefix.length);
+  }
+  return str;
+}
+
+export async function CURRENT_VERSION_STRING() {
+  const v = await getLatestVersion()
+  return v
+}
+
+export async function CURRENT_VERSION_STRING_NO_V() {
+  const v = await getLatestVersion()
+  return trimPrefix(v, "v")
+}
+
+export async function PreCodeBlock({ children, ...props }: PreProps) {
   const propsObj = { ...props }
   const propsValues = Object.values(propsObj)
   const [, , , dataLanguage, dataTheme] = propsValues
   const lang = dataLanguage || "shell"
 
-  let codeContent = ''
-  Children.forEach(children, child => {
-    if (typeof child === 'string') codeContent += child
-  })
+  let codeContent = props?.raw || ''
+
+  // If raw prop is not available, try to extract content from children
+  if (!codeContent && children) {
+    React.Children.forEach(children, (child) => {
+      if (React.isValidElement(child) && child.props.children) {
+        codeContent += child.props.children
+      }
+    })
+  }
+
+  const version = await getLatestVersion()
+  const version_no_v = trimPrefix(version, "v")
+  codeContent = codeContent.replace(/{CURRENT_VERSION}/g, version)
```

Comment:
Nice so no special release just whenever the docsite builds this parent component it'll get the info from the release registry 🚀

Reply:
Yessir! We have 2 components, and 2 replacements in each of them. One for inline code blocks, and one for entire <pre> code blocks. Then there's one replacement for the version with the `v`, and one for without the `v`